src/sentry/workflow_engine/processors/detector.py

```diff
@@ -224,35 +254,40 @@ def _extract_events_lookup_key(detector: Detector) -> int:
             )
             result.update(mapping)
 
-            missing_detector_ids = set(detector_id_to_events.keys()) - found_detector_ids
+            # events with missing detectors can be picked up by the issue stream detector
+            for project_id, events in detector_id_to_events.items():
+                if project_id not in found_detector_ids:
+                    issue_stream_events.extend(events)
+
+            missing_detector_ids.update(set(detector_id_to_events.keys()) - found_detector_ids)
 
     # Fetch detectors for events without occurrences (by project_id)
-    projects_missing_detectors = set()
+    projects_missing_detectors: set[int] = set()
     if error_events:
-        # Group events by project_id
-        project_to_events: dict[int, list[GroupEvent]] = defaultdict(list)
-
-        for event in error_events:
-            project_to_events[event.project_id].append(event)
+        mapping, projects_with_error_detectors, project_to_events = (
+            _create_default_event_detector_map(error_events, ErrorGroupType.slug)
+        )
+        result.update(mapping)
 
-        def _extract_events_lookup_key(detector: Detector) -> int:
-            return detector.project_id
+        # events with missing detectors can be picked up by the issue stream detector
+        for project_id, events in project_to_events.items():
+            if project_id not in projects_with_error_detectors:
+                issue_stream_events.extend(events)
 
-        detectors = Detector.objects.filter(
-            project_id__in=project_to_events.keys(),
-            type=ErrorGroupType.slug,
-        )
-        mapping, projects_with_error_detectors = _create_event_detector_map(
-            detectors,
-            key_event_map=project_to_events,
-            detector_key_extractor=_extract_events_lookup_key,
+    if issue_stream_events:
+        mapping, projects_with_issue_stream_detectors, project_to_events = (
+            _create_default_event_detector_map(issue_stream_events, IssueStreamGroupType.slug)
         )
         result.update(mapping)
 
-        projects_missing_detectors = set(project_to_events.keys()) - projects_with_error_detectors
+        # if no detectors, then something is wrong
```

Comment:
can we add a metric for this so we can track if anything goes wrong?

Reply:
I fixed the existing logging to capture this